plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import os
+from typing import Dict, List, Optional, Sequence
+
+from hydra import TaskFunction
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.singleton import Singleton
+from hydra.core.utils import (
+    JobReturn,
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+)
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+log = logging.getLogger(__name__)
+
+
+class SubmititLauncherDefaults(SearchPathPlugin):
+    def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
+        search_path.append(
+            "hydra-submitit-launcher",
+            "pkg://hydra_plugins.hydra_submitit_launcher.conf",
+        )
+
+
+class SubmititLauncher(Launcher):
+    def __init__(self, queue: str, folder: str, queue_parameters: DictConfig) -> None:
+        self.queue = queue
+        self.queue_parameters = queue_parameters
+        self.folder = folder
+        self.config: Optional[DictConfig] = None
+        self.config_loader: Optional[ConfigLoader] = None
+        self.task_function: Optional[TaskFunction] = None
+        self.sweep_configs: Optional[TaskFunction] = None
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ):
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+    def __call__(
+        self,
+        sweep_overrides: List[str],
+        job_dir_key: str,
+        job_num: int,
+        singleton_state: Dict[type, "Singleton"],
+    ):
+        Singleton.set_state(singleton_state)
+        configure_log(self.config.hydra.job_logging, self.config.hydra.verbose)
+        setup_globals()
+        sweep_config = self.config_loader.load_sweep_config(
+            self.config, sweep_overrides
+        )
+        with open_dict(sweep_config.hydra.job) as job:
+            # Populate new job variables
+            if "SLURM_JOB_ID" in os.environ:
+                job.id = os.environ["SLURM_JOB_ID"]
+            else:
+                job.id = "unknown"
+            sweep_config.hydra.job.num = job_num
+
+        return run_job(
+            config=sweep_config,
+            task_function=self.task_function,
+            job_dir_key=job_dir_key,
+            job_subdir_key="hydra.sweep.subdir",
+        )
+
+    def checkpoint(self, *args, **kwargs):
+        """Resubmit the current callable at its current state with the same initial arguments."""
+        # lazy import to ensure plugin discovery remains fast
+        import submitit
+
+        return submitit.helpers.DelayedSubmission(self, *args, **kwargs)
+
+    def launch(
+        self, job_overrides: Sequence[Sequence[str]], initial_job_idx: int
```

Comment:
You added initial_job_idx, right?
You also need to use it.

This also suggests that you missed a test. Look at the example launcher and be sure to introduce the tests from there.